### DISCUSSION TITLE
your instructions fail show properly how to compile plz!

### DISCUSSION THREAD

**Discussion by jamieduk** [mitmproxy/mitmproxy#7383] [Q&A]
#### Steps to reproduce the behavior:
python3 -m venv venv
venv/bin/pip install -e
Error: Command '['/home/jay/Documents/Scripts/HACKING/mitmdump/mitmproxy-main/venv/bin/python', '-m', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
bash: venv/bin/pip: No such file or directory

#### System Information
Paste the output of "mitmproxy --version" here.

**Comment by Prinzhorn** [comment]
Just to be clear, you're trying to set up local [development environment](https://github.com/mitmproxy/mitmproxy/blob/main/CONTRIBUTING.md#development-setup) using the following steps:

```sh
# 1) Verify that these commands work:
python3 --version
python3 -m pip --help
python3 -m venv --help
# 2) Install:
git clone https://github.com/mitmproxy/mitmproxy.git
cd mitmproxy
python3 -m venv venv
venv/bin/pip install -e ".[dev]"
```

Right? And the last step fails?
